### PR TITLE
Add the ability to opt-out of implicit subscription to issues on participation.

### PR DESCRIPTION
**Pull request by tkaemming**

### CODE REVIEW

File: src/sentry/api/endpoints/group_notes.py [hunk 1]
```diff
@@ -47,11 +47,18 @@ def post(self, request, group):
...
-            group=group,
+        subscribe_on_action = UserOption.objects.get_value(
             user=request.user,
-            reason=GroupSubscriptionReason.comment,
-        )
+            project=None,
+            key='subscribe_on_action',
+            default='1',
+        ) == '1'
+        if subscribe_on_action:
```
Comment (dcramer) [inline src/sentry/api/endpoints/group_notes.py]: i wonder if you'd really *not* want subscribed here?

**Review by dcramer — COMMENTED** [review]
I guess maybe if we really treat it as "automatically subscribe on participation" we shouldn't subscribe you even on assignment. The only concern is if you're assigned, in no world does not _not_ make sense to get a notification there.

File: src/sentry/api/endpoints/group_notes.py (hunk 1 above)
Comment (tkaemming) [inline src/sentry/api/endpoints/group_notes.py]: Yeah — I erred here on the side of being consistent, with any action you perform (short of subscribing) preventing you from being subscribed to an issue.

In practicality, I don't think you would want to drive-by comment on an issue without being notified about follow-up comments later. (For what it's worth, I also don't think that most people would want to change issue state, etc. without being notified about changes later, but…)

**Comment by tkaemming** [comment]
The more I think about this, the more this seems like a heavy-handed approach to fixing the workflow fatigue issue, and I'm not really sure we should do it.

**Comment by benvinegar** [comment]
Tacking on to say that we should evaluate users' response to #4471, #4472 and other related forthcoming changes before committing to this.

**Comment by benvinegar** [comment]
Close?

**Pull request closed without merging**